modules/annotation-processor/src/main/java/org/ballerinalang/natives/annotation/processor/BallerinaAnnotationProcessor.java

```diff
@@ -173,37 +172,11 @@ private void processNativeFunctions(Set<Element> balFunctionElements) {
         }
         return annot;
     }
-    
-    /**
-     * Process all {@link BallerinaConnector} annotations and append constructs to the class builder.
-     * 
-     * @param balConnectorElements Elements annotated with {@link BallerinaConnector}
-     */
-    private void processNativeConnectors(Set<Element> balConnectorElements) {
-        for (Element element : balConnectorElements) {
-            BallerinaConnector balConnector = element.getAnnotation(BallerinaConnector.class);
-            
-            /*
-             * For ballerina source modules without any native implementations, it is required to have a 
-             * dummy native impl annotation with a dummy annotation. This is a limitation in the annotation
-             * processor. Hence checking the dummy annotation here and ignore it.
-             * TODO: find a better approach to ignore annotations
-             */
-            if (IGNORE.equalsIgnoreCase(balConnector.connectorName())) {
```

Comment:
+1, moved the logic to native function. Sent the PR [1] to connectors repo.
[1] https://github.com/ballerinalang/connectors/pull/94